src/components/NftPreview/NftPreview.tsx

```diff
@@ -34,37 +36,47 @@ export const NftPreview = ({
       isSvgMediaType(getNftFilenameMediaType(nft, uri)))
   const shouldShowPlaceholder = !isString(uri) || showPlaceholder || (isUriSvg && blurRadius !== undefined) || error
 
+  useEffect(() => {
```

Comment:
is it really necessary?

Reply:
What if the nft id is set as the key of this component so that when the nft changes the component is rendered again? The loading is `true` as default an `onLoad` is set to `false`

Reply:
Not sure if it will work

Reply:
I'd say that it should receive the uri only here.

Reply:
cuz the nft info is just to figure the mimetype